src/encoder.rs

```diff
@@ -510,27 +510,27 @@ mod tests {
             D.push(C[B + i].clone());
         }
 
-        for i in 0..B {
+        for (i, c_item) in C.iter().take(B).enumerate() {
             let a = 1 + i / S;
             let b = i % S;
-            D[b] += &C[i];
+            D[b] += c_item;
 
             let b = (b + a) % S;
-            D[b] += &C[i];
+            D[b] += c_item;
 
             let b = (b + a) % S;
-            D[b] += &C[i];
+            D[b] += c_item;
         }
```

Comment:
I like the old style of the code better. Switch it back. And then disable whatever linter check this is, and disable it for the whole project by adding it in the lib.rs file